plugins/modules/files/iso_customize.py

```diff
@@ -0,0 +1,346 @@
+def iso_add_dir(module, opened_iso, iso_type, dir_path):
+    parent_dir, check_dirname = dir_path.rsplit("/", 1)
+    if not parent_dir.strip():
+        parent_dir = "/"
+    check_dirname = check_dirname.strip()
+
+    for dirname, dirlist, dummy_filelist in opened_iso.walk(iso_path=parent_dir.upper()):
+        if dirname == parent_dir.upper():
+            if check_dirname.upper() in dirlist:
+                return
+
+            if parent_dir == "/":
+                current_dirpath = "/%s" % check_dirname
+            else:
+                current_dirpath = "%s/%s" % (parent_dir, check_dirname)
+
+            current_dirpath_upper = current_dirpath.upper()
+            try:
+                if iso_type == "iso9660":
+                    opened_iso.add_directory(current_dirpath_upper)
+                elif iso_type == "rr":
+                    opened_iso.add_directory(current_dirpath_upper, rr_name=check_dirname)
+                elif iso_type == "joliet":
+                    opened_iso.add_directory(current_dirpath_upper, joliet_path=current_dirpath)
+                elif iso_type == "udf":
+                    opened_iso.add_directory(current_dirpath_upper, udf_path=current_dirpath)
+            except Exception as err:
+                msg = "Failed to create dir %s with error: %s" % (current_dirpath, to_native(err))
+                module.fail_json(msg=msg)
+
+
+def iso_add_dirs(module, opened_iso, iso_type, dir_path):
+    dirnames = dir_path.strip().split("/")
+
+    current_dirpath = "/"
+    for item in dirnames:
+        if not item.strip():
+            continue
+        if current_dirpath == "/":
+            current_dirpath = "/%s" % item
+        else:
+            current_dirpath = "%s/%s" % (current_dirpath, item)
+
+        iso_add_dir(module, opened_iso, iso_type, current_dirpath)
+
+
+def iso_check_file_exists(opened_iso, dest_file):
+    file_dir = os.path.dirname(dest_file).strip()
+    file_name = os.path.basename(dest_file)
+    dirnames = file_dir.strip().split("/")
+
+    parent_dir = "/"
+    for item in dirnames:
+        if not item.strip():
+            continue
+
+        for dirname, dirlist, dummy_filelist in opened_iso.walk(iso_path=parent_dir.upper()):
+            if dirname != parent_dir.upper():
+                break
+
+            if item.upper() not in dirlist:
+                return False
+
+        if parent_dir == "/":
+            parent_dir = "/%s" % item
+        else:
+            parent_dir = "%s/%s" % (parent_dir, item)
+
+    if '.' not in file_name:
+        file_in_iso_path = file_name.upper() + '.;1'
+    else:
+        file_in_iso_path = file_name.upper() + ';1'
+
+    for dirname, dummy_dirlist, filelist in opened_iso.walk(iso_path=parent_dir.upper()):
+        if dirname != parent_dir.upper():
+            return False
+
+        if file_name.upper() in filelist or file_in_iso_path in filelist:
+            return True
+        else:
+            return False
```

Comment:
```suggestion
        return file_name.upper() in filelist or file_in_iso_path in filelist
```

Reply:
Thanks very much. Sorry for delay fix due to the China's National Day holiday!